source/portable/BufferManagement/BufferAllocation_2.c

```diff
@@ -293,12 +296,15 @@ NetworkBufferDescriptor_t * pxGetNetworkBufferWithDescriptor( size_t xRequestedS
                      * buffer storage area, then move the buffer pointer on past the
                      * stored pointer so the pointer value is not overwritten by the
                      * application when the buffer is used. */
+                    /* MISRA Ref 11.3.1 [Misaligned access] */
+                    /* More details at: https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/main/MISRA.md#rule-113 */
+                    /* coverity[misra_c_2012_rule_11_3_violation] */
                     *( ( NetworkBufferDescriptor_t ** ) ( pxReturn->pucEthernetBuffer ) ) = pxReturn;
```

Comment:
~~~
*( ( NetworkBufferDescriptor_t ** ) ( pxReturn->pucEthernetBuffer ) ) = pxReturn;
~~~
In other places in the library, I have put asserts before assuming that a pointer can be written at a byte address. In this case, the character pointer was returned by `pvPortMalloc()`, and it is assumed that it is 4-byte aligned.

On my Xilinx UltraScale, the size of a pointer is 8 and `pvPortMalloc()` will make sure that only 64-bit aligned pointers are returned, see the `portBYTE_ALIGNMENT` macros for ARM_CA53_64_BIT:
~~~
#define portBYTE_ALIGNMENT         16
#define portBYTE_ALIGNMENT_MASK    0x000f 
~~~
In case of a 64-bit platform, the macro `ipconfigBUFFER_PADDING` will be checked with asserts: it must be at least 14, and it must be a multiple of 4 **plus** 2 bytes:
~~~
    /* This is a 64-bit platform, make sure there is enough space in
     * pucEthernetBuffer to store a pointer. */
    configASSERT( ( sizeof( uintptr_t ) != 8U ) || ( ipconfigBUFFER_PADDING >= 14 ) );

    /* But it must have this strange alignment: */
    configASSERT( ( sizeof( uintptr_t ) != 8U ) || ( ( ( ( ipconfigBUFFER_PADDING ) + 2 ) % 4 ) == 0 ) );
~~~
When `ipconfigBUFFER_PADDING` is defined, it will be assigned to `ipBUFFER_PADDING`.